composer.json

```diff
@@ -24,8 +24,14 @@
         "ext-filter": "*",
         "ext-json": "*",
         "ext-tokenizer": "*",
+        "clue/ndjson-react": "^1.0",
         "composer/semver": "^3.4",
         "composer/xdebug-handler": "^3.0.3",
+        "fidry/cpu-core-counter": "^1.0",
+        "react/child-process": "^0.6.5",
+        "react/event-loop": "^1.5",
```

Comment:
did you had a chance to consider RevoltPHP?

Reply:
No, I did not. I studied how PHPStan, Rector or ECS work and used the same approach.

Reply:
Please consider (ignore 8.1 requirement for now).
RevoltPHP is outcome of collaboration between AmPHP and ReactPHP.

If you will see it is not better to use RevoltPHP, OK for me - just want to not skip the opportunity, as creators of ReactPHP created sth newer, utilising built-in PHP 8.1+ features.

Reply:
At this point I am rather not going to change the implementation as the current one took much time from me and I just don't have enough time to re-study another library. It is a technical detail that can be changed at any point in the future, and I don't see a value in re-writing this PR as it's already working and doing what's needed. Let's not kill the killer feature because of other possibilities available 🙂.

Reply:
I do fail to understand where i was even remotely close to "kill the feature" with question about selection of tool or explicitly saying "OK for me" (to keep using whatever is in use)

Reply:
I fail to understand why you asked about other library which requires PHP 8.1 when there's working implementation that does not force us to drop support for 2 PHP versions 😉. Anyway, I do not consider switching to RevoltPHP at this point.

Reply:
It's not only that, RevoltPHP is only about the event loop. You'd still need a `child process` and `socket` package to handle those parts of this PR.

> creators of ReactPHP created sth newer, utilising built-in PHP 8.1+ features.

Yup, `react/async` v4, which is out of scope for this PR honestly.

Reply:
I asked because creators of one library created sth new and I was wondering if it can give us benefits. If not, all good 🤷🏻 
(and 8.1+ req we could overcome, no worry ;P wanted to see if topic is relevant before potentially solving 8.1+)

Reply:
> I asked because creators of one library created sth new and I was wondering if it can give us benefits. If not, all good 🤷🏻 

Much appreciated :heart: , that is also why I added a bit more context.

> (and 8.1+ req we could overcome, no worry ;P wanted to see if topic is relevant before potentially solving 8.1+)

Please ping me if that becomes a discussion/thing. We're starting to work on our next major version which will cover that topic. (I don't want to hijack this PR for that in anyway.)

Reply:
You can use the RevoltPHP event loop and have compatibility with ReactPHP + AMPHP libraries. Using the ReactPHP event loop locks you into only ReactPHP libraries. This does come at the cost of requiring PHP 8.1+, so it is something which could be changed in the future with little to no effort.

Reply:
Thank you @trowski for the info. That's how I see it - for now it's better to stick with ReactPHP's event loop because I don't want to enforce dropping 2 PHP versions with this change 🙂.